tensorflow/contrib/seq2seq/python/ops/beam_search_decoder.py

```diff
@@ -189,6 +189,45 @@ def _check_maybe(t):
     raise ValueError(
         "Expected tensor (%s) to have known rank, but ndims == None." % t)
 
+def _check_static_batch_beam_maybe(shape, batch_size, beam_width):
+  """Raises an exception if dimensions are known statically and can not be
+  reshaped to [batch_size, beam_size, -1].
+  """
+  reshaped_shape = tensor_shape.TensorShape([batch_size, beam_width, None])
+  if (batch_size is not None and shape[0].value is not None
+      and (shape[0] != batch_size * beam_width
+           or (shape.ndims >= 2 and shape[1].value is not None
+               and (shape[0] != batch_size or shape[1] != beam_width)))):
+    raise ValueError("TensorArray reordering expects elements to be "
+                     "reshapable to %s which is incompatible with the "
+                     "current shape %s. Consider setting "
+                     "reorder_tensor_arrays to False to disable TensorArray "
+                     "reordering during the beam search."
+                     % (reshaped_shape, shape))
+
+def _check_batch_beam(t, batch_size, beam_width):
+  """Returns an Assert operation checking that the elements of the stacked
+  TensorArray can be reshaped to [batch_size, beam_size, -1].
+  """
+  error_message = ("TensorArray reordering expects elements to be "
+                   "reshapable to [batch_size, beam_size, -1] which is "
+                   "incompatible with the dynamic shape of %s elements. "
+                   "Consider setting reorder_tensor_arrays to False to disable "
+                   "TensorArray reordering during the beam search."
+                   % (t.name))
+  return control_flow_ops.Assert(
+      math_ops.logical_and(
+          math_ops.greater_equal(array_ops.rank(t), 2),
```

Comment:
here for ease of reading, you can use:

array_ops.rank(t) >= 2

below you can use:

array_ops.rank(t) > 2